### DISCUSSION TITLE
Css Minimizer plugin issue with Tailwind CSS

### DISCUSSION THREAD

**Discussion by symmetrically** [tailwindlabs/tailwindcss#7858] [Help]
I am using create-react-app version 5 to do a production build. Here is the error I am getting:

```shell

❯ npm run build

> create-react-app@0.1.0 build
> react-scripts build

Creating an optimized production build...
Failed to compile.

static/css/main.8b7f8789.css from Css Minimizer plugin
/path/to/reactapp/static/css/main.8b7f8789.css:990:19: Unknown word [<no source>:1,0][static/css/main.8b7f8789.css:990,19]

```
The CSS file looks like this:

``` CSS
@tailwind base;
@tailwind components;
@tailwind utilities;

/* Custom CSS here */

```
Build succeeds if I remove the tailwind directives from the CSS file. 

Another user is also experiencing the same problem and asked for help on [stackoverflow](https://stackoverflow.com/questions/71484883/unknown-word-error-from-css-minimizer-plugin-on-react-build) 

One interesting thing is if I run:

```shell
❯ npx tailwindcss -i src/index.css -o dist/output.css

Done in 674ms.
```

The output.css file contains following which does not look right:

```css
 .aspect-\[\$\{videoRatio\}\] {
   aspect-ratio: ${videoRatio};
}
```

Any help to solve this problem would be appreciated.

Node version: 16.13.0
OS: macOS Monterey

**Comment by symmetrically** [comment]
Can I get some help here please ?

**Comment by jonnitto** [comment]
I looks like you are trying to use dynamic values. Note that you still need to [write purgeable HTML](https://tailwindcss.com/docs/optimizing-for-production#writing-purgeable-html) when using arbitrary values, and your classes need to exist as complete strings for Tailwind to detect them correctly.

### Don't use string concatenation to create class names
```html
<div className={`aspect-[${videoRatio}]`}></div>
```

### Do dynamically select a complete class name
```
<div className={ videoRatio === '16/9' ? 'aspect-[16/9]' : 'aspect-[4/3]' }></div>
```

Tailwind doesn’t include any sort of client-side runtime, so class names need to be statically extractable at build-time, and can’t depend on any sort of arbitrary dynamic values that change on the client. Use inline styles for these situations, or combine Tailwind with a CSS-in-JS library like [Emotion](https://emotion.sh/docs/introduction) if it makes sense for your project.

### Use inline styles for truly dynamic or user-defined values
```html
<div style={`aspect-ratio: ${videoRatio}`}></div>
```
